pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/RulesToExecute.java

```diff
@@ -160,9 +169,14 @@ public boolean isRecommendRealtimeProvisioning() {
     return _recommendRealtimeProvisioning;
   }
 
+  public boolean isRecommendSegmentSize() {
+    return _recommendSegmentSize;
+  }
+
   // Be careful with the sequence, each rule can execute individually
   // but a rule may depend on its previous rule when they both fired
   public enum Rule {
+    SegmentSizeRule, // This rule must be the first rule. It provides segment count, segment size, numRows in segments which are used in other rules
```

Comment:
(nit) might want to add it also derives CPS.

Reply:
Sure.